addons/mrp/models/product.py

```diff
@@ -50,6 +50,15 @@ def _compute_is_kits(self):
         for template in self:
             template.is_kits = (template.id in kits_ids)
 
+    def _search_is_kits(self, operator, value):
+        assert operator in ('=', '!='), 'Unsupported operator'
+        bom_tmpl = self.env['mrp.bom'].sudo()._search(
```

Comment:
it's a query, it's better that the name reflects that
also, why sudo? We try to avoid using sudo as much as possible. In this case, if you have products that are not in a single company and have a kit bom in company A but not company B, searching in company B will still show the products.
```suggestion
        bom_query = self.env['mrp.bom']._search(
```